### PR TITLE
Add validation to request body in controllers

### LINKED ISSUE #86: Add RequestBody Validation in controllers
There is 500 status code when I'm requesting wrong request body instead of 400 status code.

### PR DESCRIPTION
**Pull request by MuchosChuchos**
Closes #86

### CODE REVIEW

File: api/src/main/java/com/eleks/academy/pharmagator/dataproviders/dto/input/MedicineDto.java [hunk 1]
```diff
@@ -2,7 +2,12 @@
 
 import lombok.Data;
 
+import javax.validation.constraints.NotNull;
+
 @Data
 public class MedicineDto {
+
+    @NotNull
```
Comment (zyoneal) [inline api/src/main/java/com/eleks/academy/pharmagator/dataproviders/dto/input/MedicineDto.java:10]: I think NotBlank is better than NotNull. How do you think?

File: api/src/main/java/com/eleks/academy/pharmagator/dataproviders/dto/input/PharmacyDto.java [hunk 2]
```diff
@@ -2,8 +2,14 @@
 
 import lombok.Data;
 
+import javax.validation.constraints.NotNull;
+
 @Data
 public class PharmacyDto {
+
+    @NotNull
```
Comment (zyoneal) [inline api/src/main/java/com/eleks/academy/pharmagator/dataproviders/dto/input/PharmacyDto.java:10]: same

File: api/src/main/java/com/eleks/academy/pharmagator/dataproviders/dto/input/MedicineDto.java (hunk 1 above)
Comment (MuchosChuchos) [reply api/src/main/java/com/eleks/academy/pharmagator/dataproviders/dto/input/MedicineDto.java:10]: I think the same as you ^-^

File: api/src/main/java/com/eleks/academy/pharmagator/dataproviders/dto/input/PharmacyDto.java (hunk 2 above)
Comment (MuchosChuchos) [reply api/src/main/java/com/eleks/academy/pharmagator/dataproviders/dto/input/PharmacyDto.java]: done

File: api/src/main/java/com/eleks/academy/pharmagator/dataproviders/dto/input/MedicineDto.java (hunk 1 above)
Comment (MuchosChuchos) [reply api/src/main/java/com/eleks/academy/pharmagator/dataproviders/dto/input/MedicineDto.java]: done

**Review by zyoneal — DISMISSED** [review]

File: api/src/main/java/com/eleks/academy/pharmagator/dataproviders/dto/input/PriceDto.java [hunk 3]
```diff
@@ -3,14 +3,17 @@
...
 
 import javax.validation.constraints.Min;
+import javax.validation.constraints.NotBlank;
 import java.math.BigDecimal;
 
 @Data
 public class PriceDto {
 
     @Min(value = 0)
+    @NotBlank
```
Comment (pavlosloboda) [inline api/src/main/java/com/eleks/academy/pharmagator/dataproviders/dto/input/PriceDto.java:13]: `@NotBlank` does not make sense for `BigDecimal`

Comment (MuchosChuchos) [reply api/src/main/java/com/eleks/academy/pharmagator/dataproviders/dto/input/PriceDto.java]: solved

File: api/src/main/java/com/eleks/academy/pharmagator/dataproviders/dto/input/MedicineDto.java [hunk 4]
```diff
@@ -2,7 +2,12 @@
 
 import lombok.Data;
 
+import javax.validation.constraints.NotBlank;
+
 @Data
 public class MedicineDto {
```
Comment (sergvasylchak) [inline api/src/main/java/com/eleks/academy/pharmagator/dataproviders/dto/input/MedicineDto.java:8]: `@NoArgs...` + `@AllArgs...`
Same for other request bodies

Comment (MuchosChuchos) [reply api/src/main/java/com/eleks/academy/pharmagator/dataproviders/dto/input/MedicineDto.java:12]: Done

**Review by Oleksandr-Eleks — APPROVED** [review]

**Review by Oleksandr-Eleks — APPROVED** [review]

**Review by Zond47 — APPROVED** [review]

**Review by sergvasylchak — APPROVED** [review]

**Pull request merged**